examples/pna-example-exactmap-ternarymap.p4

```diff
@@ -0,0 +1,327 @@
+#include "../pna.p4"
+
+
+// Very simple PNA program intended only to demonstrate how to send
+// unicast packets to a destination network port, or host vport, based
+// upon the destination MAC address in the Ethernet header.
+
+
+typedef bit<48>  EthernetAddress;
+
+header ethernet_t {
+    EthernetAddress dstAddr;
+    EthernetAddress srcAddr;
+    bit<16>         etherType;
+}
+
+//////////////////////////////////////////////////////////////////////
+// Struct types for holding user-defined collections of headers and
+// metadata in the P4 developer's program.
+//////////////////////////////////////////////////////////////////////
+
+struct metadata_t {
+}
+
+struct headers_t {
+    ethernet_t eth;
+}
+
+struct emap1_key {
+    bit<8> my_field;
+}
+
+struct emap2_key {
+    bit<4> k1;
+    bit<9> k2;
+}
+
+struct emap2_val {
+    bit<4>  val1;
+    bit<12> val2;
+}
+
+struct tmap1_key {
+    bit<24> f1;
+    bit<3> f2;
+}
+
+struct tmap2_key {
+    bit<18> only_field;
+}
+
+struct tmap2_val {
+    bit<8>  v1;
+    bit<10> v2;
+}
+
+parser MainParserImpl(
+    packet_in pkt,
+    out   headers_t  hdr,
+    inout metadata_t meta,
+    in    pna_main_parser_input_metadata_t istd)
+{
+    state start {
+        pkt.extract(hdr.eth);
+        transition accept;
+    }
+}
+
+control MainControlImpl(
+    inout headers_t  hdr,
+    inout metadata_t meta,
+    in    pna_main_input_metadata_t  istd,
+    inout pna_main_output_metadata_t ostd)
+{
+
+    // Example instantiation from documentation comments for ExactMap
+    // extern in pna.p4
+/*
+    The control plane API for ExactMap emap1 is exactly the same as
+    for the following P4 table.  The action name `lookup` is
+    hard-coded into the P4 compiler as the action name for _all_
+    instances of ExactMap and TernaryMap externs.
+
+    // Note that the name 'lookup_value' is a default action parameter
+    // name generated by the P4 compiler in this case, because the
+    // value type of emap1 is bit<16>, with no name.
+    action lookup (bit<16> lookup_value) {
+        // body of action is irrelevant for control plane API generation
+    }
+    table emap1 {
+        key = {
+            // The key fields are all of the fields of struct
+            // emap1_key, with the same names as the member field
+            // names in the declaration of struct emap1_key.  All have
+            // match_kind exact, because emap1 is ExactMap, not
+            // TernaryMap.
+            my_field: exact;
+        }
+        actions = { lookup; }
+        default_action = lookup(42);
+    }
+
+    Note that as of 2022-Nov, the open source p4c front end gives a
+    compile time error if there is a table and an extern instance
+    defined within a control that have the same name as each other.
+    Thus we can generate control plane API files, e.g. P4Info files,
+    assuming that extern instances of ExactMap and TernaryMap will
+    never have the same names as any P4 tables in the same control.
```

Comment:
@jamescchoi See the paragraph comment I just added earlier today (2022-Nov-29) following up on a question in a meeting we had earlier today.